swift-sdk/Internal/InAppManager.swift

```diff
@@ -297,16 +375,29 @@ extension InAppManager : InAppSynchronizerDelegate {
         }
     }
     
-    private func getFirstProcessableMessage() -> IterableInAppMessage? {
-        return messagesMap.values.filter({ $0.processed == false && $0.trigger.type == .immediate }).first
+    private func getFirstProcessableInAppMessage() -> IterableInAppMessage? {
+        return messagesMap.values.filter(InAppManager.isProcessable).first
+    }
+    
+    private static func isProcessable(message: IterableInAppMessage) -> Bool {
+        return message.didProcessTrigger == false && message.trigger.type == .immediate
     }
     
-    private func updateMessage(_ message: IterableInAppMessage, processed: Bool, consumed: Bool = false) {
+    private func updateMessage(_ message: IterableInAppMessage,
+                               didProcessTrigger: Bool? = false,
+                               didProcessInbox: Bool? = false,
+                               consumed: Bool = false) {
         ITBDebug()
         updateQueue.sync {
-            message.processed = processed
-            message.consumed = consumed
-            self.messagesMap.updateValue(message, forKey: message.messageId)
+            let toUpdate = message
+            if let didProcessTrigger = didProcessTrigger {
+                toUpdate.didProcessTrigger = didProcessTrigger
+            }
+            if let didProcessInbox = didProcessInbox {
```

Comment:
Can we remove `didProcessInbox`? We don't stop processing them like in-app messages, and we probably know if there are changes when comparing the remote queue with the local queue, and can trigger updates based on that.